src/simulation.hpp

```diff
@@ -1369,6 +1505,40 @@ void AMRSimulation<problem_t>::GetData(int lev, amrex::Real time,
   }
 }
 
+// utility to copy in data from state_old_fc_[lev][dir] and/or state_new_fc_[lev][dir]
+// into another multifab
+template <typename problem_t>
+void AMRSimulation<problem_t>::GetFCData(int lev, amrex::Real time,
+                                         amrex::Vector<amrex::MultiFab *> &data,
+                                         amrex::Vector<amrex::Real> &datatime,
+                                         quokka::direction dir) {
+  BL_PROFILE("AMRSimulation::GetFCData()");
+
+  data.clear();
+  datatime.clear();
+
+  const amrex::Real teps =
+      (tNew_[lev] - tOld_[lev]) * 1.e-3; // generous roundoff error threshold
+
+  int dim = static_cast<int>(dir);
+
+  if (time > tNew_[lev] - teps &&
+      time < tNew_[lev] + teps) { // if time == tNew_[lev] within roundoff
+    data.push_back(&state_new_fc_[lev][dim]);
```

Comment:
The issue is that we want to avoid code duplication between the cell-centered and face-centered version of this function. Currently, the face-centered and cell-centered versions actually have different behavior, since I have updated the roundoff criteria for the cell-centered version since you started working on this pull request.